src/aws/osml/image_processing/gdal_tile_factory.py

```diff
@@ -339,31 +339,85 @@ def _read_from_rlevel_as_array(
 
         return result
 
-    def _create_display_image(self, pixel_array: np.array) -> np.array:
+    def _normalize_image(self, pixel_array: np.array) -> np.array:
         """
-        This method selects the first 3 bands of a multi-band image (or a single band of grayscale) and performs a
-        simple dynamic range adjustment to map those values to the 0-255 range of an 8-bit per pixel image for
-        visualization.
+        This method applies the specified range adjustment to the requested image and returns the adjusted pixels. If
+        the range adjustment is set to NONE it returns the original pixels.
 
         :param pixel_array: the input image pixels
-        :return: a range adjusted 8-bit per pixel image of up to 3 bands
+        :return: a range adjusted 8-bit per pixel image
         """
-        if pixel_array.ndim == 3 and pixel_array.shape[2] > 3:
+        if self.range_adjustment == RangeAdjustmentType.DRA:
+            return self._normalize_bands(pixel_array, self._normalize_band_dra)
+        elif self.range_adjustment == RangeAdjustmentType.MINMAX:
+            return self._normalize_bands(pixel_array, self._normalize_band_minmax)
+        else:
+            return pixel_array
```

Comment:
With this change I'd like to see some logging of images that OpenCV2 fails to encode into the output type. Somewhere around line 294 in the create_orthophoto_tile function. 

The issue is that a range adjustment will be necessary to reduce the bit-range of a lot of images to comply with limits of common commercial formats. This change is fine in that maybe someone does want the full original image bit depth and some encodings can handle it but others might simply fail. Instead of just silently returning None adding some debug logging that tracks the fact that OpenCV failed to encode the image might save us some time in the future.

Reply:
Will update to force pixels to 8-bit to avoid this case since the methods are specific to generating ortho map tiles.  We can extend them later as needed.